src/cgame/cg_gameinfo.cpp

```diff
@@ -31,12 +31,195 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #include "cg_local.h"
 
+//
+// arena info
+//
+static int  cg_numArenas;
+static char *cg_arenaInfos[ MAX_ARENAS ];
+
+/*
+===============
+CG_ParseInfos
+===============
+*/
+static int CG_ParseInfos( const char *buf, int max, char *infos[] )
+{
+	char *token;
+	int  count = 0;
+	char key[ MAX_TOKEN_CHARS ];
+	char info[ MAX_INFO_STRING ];
+
+	auto infoPostfixLen = strlen( "\\num\\" ) + strlen( va( "%d", MAX_ARENAS ) );
+
+	while ( 1 )
+	{
+		token = COM_Parse( &buf );
+
+		if ( !token[ 0 ] )
+		{
+			break;
+		}
+
+		if ( strcmp( token, "{" ) )
+		{
+			Log::Notice( "Missing { in info file\n" );
+			break;
+		}
+
+		if ( count == max )
+		{
+			Log::Notice( "Max infos exceeded\n" );
+			break;
+		}
+
+		info[ 0 ] = '\0';
+
+		while ( 1 )
+		{
+			token = COM_ParseExt( &buf, true );
+
+			if ( !token[ 0 ] )
+			{
+				Log::Notice( "Unexpected end of info file\n" );
+				break;
+			}
+
+			if ( !strcmp( token, "}" ) )
+			{
+				break;
+			}
+
+			Q_strncpyz( key, token, sizeof( key ) );
+
+			token = COM_ParseExt( &buf, false );
+
+			if ( !token[ 0 ] )
+			{
+				strcpy( token, "<NULL>" );
+			}
+
+			Info_SetValueForKey( info, key, token, false );
+		}
+
+		//NOTE: extra space for arena number
+		infos[ count ] = (char*) BG_Alloc( strlen( info ) + infoPostfixLen + 1 );
+
+		if ( infos[ count ] )
+		{
+			strcpy( infos[ count ], info );
+			count++;
+		}
+	}
+
+	return count;
+}
+
+/*
+===============
+CG_LoadArenasFromFile
+===============
+*/
+static int CG_LoadArenasFromFile( const char *filename )
+{
+	std::error_code err;
+	std::string text = FS::PakPath::ReadFile( filename, err );
+	if ( err )
+	{
+		Log::Warn( "file not found: %s", filename );
+		return 0;
+	}
+
+	if ( text.size() >= MAX_ARENAS_TEXT )
+	{
+		Log::Warn( "file too large: %s is %i, max allowed is %i",
+			filename, text.size(), MAX_ARENAS_TEXT );
+
+		/* Tell it was found but not loaded properly,
+		it makes possible to not look for legacy file
+		that should not exist when the non-legacy file
+		actuallyexists, while not losing the information
+		and not returning true to not deceive futur users
+		of this function. */
+		return -1;
+	}
+
+	cg_numArenas += CG_ParseInfos( text.c_str(), MAX_ARENAS - cg_numArenas, &cg_arenaInfos[ cg_numArenas ] );
+
+	return 1;
+}
+
+/*
+===============
+CG_MapLoadNameCompare
+===============
+*/
+static int CG_MapLoadNameCompare( const void *a, const void *b )
+{
+	arenaInfo_t *A = ( arenaInfo_t * ) a;
+	arenaInfo_t *B = ( arenaInfo_t * ) b;
+
+	return Q_stricmp( A->name, B->name );
+}
+
 /*
 ===============
 CG_LoadArenas
 ===============
 */
-void CG_LoadArenas()
+void CG_LoadArenas( std::string mapname )
+{
+	cg_numArenas = 0;
+	rocketInfo.data.arenaCount = 0;
+
+	std::string arena_file = Str::Format( "meta/%s/%s.arena", mapname, mapname );
+
+	Log::Debug( "Loading arena file: %s", arena_file );
+
+	if ( CG_LoadArenasFromFile( arena_file.c_str() ) == 0 )
+	{
+		Log::Warn( "Arena file not found: %s", arena_file );
```

Comment:
You have duplicated file not found warnings here and inside CG_LoadArenasFromFile

Reply:
Removed.